jane/doc/extensions/_05-kinds/syntax.md

```diff
@@ -113,30 +114,29 @@ The abbreviations defined in the language are as follows:
       unyielding stateless` (all of which only affect functions).
     * ... support mutation: this allows them to mode-cross to `contended
       immutable`.
-    * ... equal the bit pattern comprising all `0`s: this allows a bound
-      of `non_null`.
     * ... point anywhere: this allows mode-crossing to `aliased` (which would
-      confuse update-in-place of the pointed-to memory, if there was any) and a
-      bound of `non_float` (because the value is not a pointer to a
-      floating-point block).
-
+      confuse update-in-place of the pointed-to memory, if there was any)
     Using `mod everything` is appropriate for data represented directly,
     like `int` or `float32#`.
 
-* `any_non_null = any mod non_null`
+* `any_non_null = any mod non_null separable`
 
-* `value = value_or_null mod non_null`
+* `value = value_or_null mod non_null separable`
 
     This is the kind of typical OCaml values, as they have been before OxCaml
     was invented. When a kind is unknown (but `any` would not be an appropriate
     choice), we default kinds to be `value`. This defaulting action is described
     in further detail in the sections below.
 
-* `immediate = value mod everything`
+* `immediate = value mod everything non_null non_float`
 
     This is the kind of `int` and types like it (including enumerations and
     `bool`). It is the OxCaml equivalent of OCaml's `[@@immediate]` attribute.
 
+* `immediate_or_null = value_or_null mod everything`
```

Comment:
Shouldn't this still be `non_float`?

Reply:
Should be, but for now it's the jkind of `or_null`, and since with-bounds don't affect separability I gave it the most conservative value.